src/openapi-generator.ts

```diff
@@ -620,6 +625,23 @@ export class OpenAPIGenerator {
     };
   }
 
+  private mapNullableType(
+    type: NonNullable<SchemaObject['type']>,
+    isNullable: boolean
+  ): Pick<SchemaObject, 'type' | 'nullable'> {
+    // Open API 3.1.0 made the `nullable` key invalid and instead you use type arrays
+    if (isNullable && this.openAPIVersion && this.openAPIVersion >= '3.1.0') {
```

Comment:
Using an enum we would be able to specify exact behaviors. Comparing version strings should not be done like that. For example `'10.1.1' > '3.1.0'` would evaluate to `false` (obviously that's way ahead of time but it is an example). There is a [library](https://www.npmjs.com/package/compare-versions) for that but if we can implement it ourselves utilizing enums I think that would be better.

Reply:
Ahhhh yeah stupid me 🫣

Reply:
I'm not sure if what I've done is what you had in mind, if we want to allow unspecified/newer versions I can always rip their `satisfies` code out into our repo 😄

Reply:
Yup using a simple array is what I was thinking of as well